tests/js/spec/views/settings/organizationIntegrations/sentryAppInstallations.spec.jsx

```diff
@@ -0,0 +1,185 @@
+/*global global*/
+import React from 'react';
+
+import {Client} from 'app/api';
+import {mount} from 'enzyme';
+import SentryAppInstallations from 'app/views/organizationIntegrations/sentryAppInstallations';
+
+describe('Sentry App Installations', function() {
+  let org = TestStubs.Organization();
+  let sentryApp = TestStubs.SentryApp();
+  let install = TestStubs.SentryAppInstallation({
+    organization: {slug: org.slug},
+    app: {slug: sentryApp.slug, uuid: 'f4d972ba-1177-4974-943e-4800fe8c7d05'},
+    code: '50624ecb-7aac-49d6-934a-83e53677560f',
+  });
+
+  let routerContext = TestStubs.routerContext();
+
+  beforeEach(() => {
+    Client.clearMockResponses();
+  });
+
+  describe('when no Apps exist', () => {
+    Client.addMockResponse({
+      url: `/organizations/${org.slug}/sentry-apps/`,
+      body: [],
+    });
+
+    Client.addMockResponse({
+      url: `/organizations/${org.slug}/sentry-app-installations/`,
+      body: [],
+    });
+
+    const wrapper = mount(<SentryAppInstallations orgId={org.slug} />, routerContext);
+
+    it('no row is displayed', () => {
+      expect(wrapper).toMatchSnapshot();
+      expect(wrapper.exists('SentryApplicationRow')).toBe(false);
+    });
+  });
+
+  describe('when Apps exist', () => {
+    Client.addMockResponse({
+      url: `/organizations/${org.slug}/sentry-apps/`,
+      body: [sentryApp],
+    });
+
+    Client.addMockResponse({
+      url: `/organizations/${org.slug}/sentry-app-installations/`,
+      body: [],
+    });
+
+    let wrapper = mount(<SentryAppInstallations orgId={org.slug} />, routerContext);
+
+    it('displays all Apps owned by the Org', () => {
+      expect(wrapper).toMatchSnapshot();
+      expect(wrapper.find('SentryApplicationRow').prop('app').name).toBe('Sample App');
+    });
+
+    describe('when installing', () => {
+      beforeEach(() => {
+        Client.addMockResponse({
+          url: `/organizations/${org.slug}/sentry-app-installations/`,
+          method: 'POST',
+          body: install,
+        });
+      });
+
+      it('disallows installation when already installed', () => {
+        Client.addMockResponse({
+          url: `/organizations/${org.slug}/sentry-apps/`,
+          method: 'GET',
+          body: [sentryApp],
+        });
+
+        Client.addMockResponse({
+          url: `/organizations/${org.slug}/sentry-app-installations/`,
+          body: [install],
+          method: 'GET',
+        });
+
+        wrapper = mount(<SentryAppInstallations orgId={org.slug} />, routerContext);
+
+        expect(wrapper.find('[icon="icon-trash"]').exists()).toBe(true);
+      });
+
+      it('redirects the user to the Integrations page when a redirectUrl is not set', () => {
```

Comment:
Should you have tests for when the api calls fail?

Reply:
Do you mean to check to make sure the success message doesn't show if it fails?